pkg/store/store.go

```diff
@@ -160,6 +168,100 @@ type Store struct {
 	numTrailingLogs uint64
 }
 
+// lastCommittedIndex get the last committed index in raft log
+func (s *Store) lastCommittedIndex() (uint64, bool) {
+	if s.State() != Leader {
+		return 0, false
+	}
+
+	readIndex := s.raft.LastIndex()
+	if s.raft.VerifyLeader().Error() != nil {
+		return 0, false // leader transfer occurs
+	}
+	return readIndex, true
+}
+
+// queryLeaderForReadIndex query read index from leader
+func (s *Store) queryLeaderForReadIndex() (uint64, bool) {
+	if s.State() == Leader {
+		return s.lastCommittedIndex()
+	}
+
+	conn, err := net.Dial("tcp", s.LeaderAddr())
```

Comment:
We can create a client to request the HTTP server or GRPC server.

Reply:
My original goal is to reuse the port. Is it necessary to use another port to serve?

Reply:
I suppose this is acceptable, and it’s lighter than gRPC and HTTP. For now, in-node communications, I don't think it benefits from using gRPC or HTTP. Maybe in the future, we can refactor this part by using gRPC. cc @noneback

Reply:
OK, in-node is lightweight.